pkg/virt-launcher/virtwrap/device/hostdevice/gpu/hostdev.go

```diff
@@ -79,9 +221,23 @@ func createHostDevicesMetadata(vmiGPUs []v1.GPU) []hostdevice.HostDeviceMetaData
 // The validation assumes that the assignment of a device to a specified GPU is correct,
 // therefore a simple quantity check is sufficient.
 func validateCreationOfAllDevices(gpus []v1.GPU, hostDevices []api.HostDevice) error {
-	if len(gpus) != len(hostDevices) {
+	gpusWithDP := []v1.GPU{}
+	gpusWithDRA := []v1.GPU{}
+	for _, gpu := range gpus {
+		if gpu.ClaimRequest != nil {
+			gpusWithDRA = append(gpusWithDRA, gpu)
+		} else {
+			gpusWithDP = append(gpusWithDP, gpu)
+		}
+	}
+
+	if len(gpusWithDRA) > 0 && len(gpusWithDP) > 0 {
+		return fmt.Errorf("invalid VMI spec, both DRA and device plugin GPUs are not supported at the same time")
```

Comment:
Aren't they? why? was that discussed on the VEP?

Reply:
Allowing both device plugins and DRA is not a solved problem in k8s. How can a gpu-dp driver and dra driver both manage the same device on a node? This is unexplored territory, for safety it is best to not support such a usecase

Reply:
Huh, that's new to me, thanks for bringing this up.

> How can a gpu-dp driver and dra driver both manage the same device on a node?

Please correct me if I'm wrong, but doesn't this depend on the specific setup?
For example, let's say that I have a host with GPUs from two different vendors, say Nvidia and AMD, and I want to consume both in the same VMI, the Nvidia GPU through DRA and AMD through a device plugin. The example also applies for a host device.

Does this example make sense?
Is it restricted in k8s for "regular" pods?

Reply:
@iholder101 I agree.
@alaypatel07, what about a device plugin and driver managing different devices on a node?
If I remember correctly, we discussed this during the design and said that if there isn't a template, we fall back to device plugins.

Reply:
removed the check.

Reply:
@alaypatel07, also think about the testing/adoption path. Users who already have running VMs with device plugins want to try out DRA for other devices will not be able to do so.